src/main/java/net/dv8tion/jda/api/utils/messages/MessageEditData.java

```diff
@@ -0,0 +1,343 @@
+     */
+    @Nonnull
+    public List<LayoutComponent> getComponents()
+    {
+        return components;
+    }
+
+    /**
+     * The {@link AttachedFile AttachedFiles} attached to this message.
+     *
+     * @return The list of attachments, or an empty list if none were set
+     */
+    @Nonnull
+    public List<AttachedFile> getAttachments()
+    {
+        return files;
+    }
+
+    /**
+     * The IDs for users which are allowed to be mentioned, or an empty list.
+     *
+     * @return The user IDs which are mention whitelisted
+     */
+    @Nonnull
+    public Set<String> getMentionedUsers()
+    {
+        return allowedMentions.getUsers();
+    }
+
+    /**
+     * The IDs for roles which are allowed to be mentioned, or an empty list.
+     *
+     * @return The role IDs which are mention whitelisted
+     */
+    @Nonnull
+    public Set<String> getMentionedRoles()
+    {
+        return allowedMentions.getRoles();
+    }
+
+    /**
+     * The mention types which are whitelisted.
+     *
+     * @return The mention types which can be mentioned by this message
+     */
+    @Nonnull
+    public EnumSet<Message.MentionType> getAllowedMentions()
+    {
+        return allowedMentions.getAllowedMentions();
+    }
+
+    /**
+     * Whether this message would mention a user, if it is sent as a reply.
+     *
+     * @return True, if this would mention with the reply
+     */
+    public boolean isMentionRepliedUser()
+    {
+        return allowedMentions.isMentionRepliedUser();
+    }
+
+    @Nonnull
+    @Override
+    public synchronized DataObject toData()
+    {
+        DataObject json = DataObject.empty();
+        if (isSet(CONTENT))
+            json.put("content", content);
+        if (isSet(EMBEDS))
+            json.put("embeds", DataArray.fromCollection(embeds));
+        if (isSet(COMPONENTS))
+            json.put("components", DataArray.fromCollection(components));
+        if (isSet(MENTIONS))
+            json.put("allowed_mentions", allowedMentions);
+        if (isSet(ATTACHMENTS))
+        {
+            DataArray attachments = DataArray.empty();
+            json.put("attachments", attachments);
+            for (int i = 0; i < files.size(); i++)
+                attachments.add(files.get(i).toAttachmentData(i));
+        }
+
+        return json;
+    }
+
+    /**
+     * The {@link FileUpload FileUploads} attached to this message.
+     *
+     * @return The list of file uploads
+     */
+    @Nonnull
+    public synchronized List<FileUpload> getFiles()
+    {
+        return Collections.unmodifiableList(
+            files.stream()
+                 .filter(FileUpload.class::isInstance)
+                 .map(FileUpload.class::cast)
+                 .collect(Collectors.toList())
+        );
+    }
+
+    @Override
+    public synchronized void close()
+    {
+        files.forEach(IOUtil::silentClose);
+    }
+
+    private boolean isSet(int flag)
```

Comment:
Lets make this package protected so that siblings can call it.

Reply:
Honestly though, I do wonder if it would be useful for users to have access to this too 🤔